src/BuildSqlGroupByClauseCapableTrait.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace RebelCode\Storage\Resource\Sql;
+
+use Dhii\Exception\InternalExceptionInterface;
+use Dhii\Storage\Resource\Sql\EntityFieldInterface;
+use Dhii\Util\String\StringableInterface as Stringable;
+use Exception as RootException;
+use InvalidArgumentException;
+use OutOfBoundsException;
+use OutOfRangeException;
+use stdClass;
+use Traversable;
+
+/**
+ * Common functionality for building GROUP BY query portions.
+ *
+ * @since [*next-version*]
+ */
+trait BuildSqlGroupByClauseCapableTrait
+{
+    /**
+     * Builds the GROUP BY portion of the an SQL query.
+     *
+     * @since [*next-version*]
+     *
+     * @param array|stdClass|Traversable $grouping A list of strings, stringable objects or entity-field instances.
```

Comment:
Instead of generic `array`, please add typed arrays of actual allowed element types.

Reply:
Ok